conda/resolve.py

```diff
@@ -324,6 +335,7 @@ def _broader(self, ms, specs_by_name):
             return True
         return False
 
+    @memoizemethod
```

Comment:
`get_reduced_index` has a cache inside the function based on strict_channel priority and a frozen set of the spec argument.  Does the addition of the `memorizemethod` improve the performance significantly?

Reply:
Good point. Yes, this seems to have a large effect. Try it and see for yourself, though. Maybe the key for the existing cache is overly specific.